osu.Game/Screens/Edit/Screens/Compose/Layers/DragBox.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input;
+using osu.Game.Rulesets.Edit;
+using OpenTK.Graphics;
+
+namespace osu.Game.Screens.Edit.Screens.Compose.Layers
+{
+    /// <summary>
+    /// A box that handles and displays drag selection for a collection of <see cref="HitObjectMask"/>s.
+    /// </summary>
+    public class DragBox : CompositeDrawable

```

Comment:
Should probably be called something like DragLayer, as it itself is not a box but contains a box.